### PR TITLE
ENT-3787: Swaps to GitHub actions

### PR DESCRIPTION
**Pull request by jlajoie**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

**Comment by UsamaSadiq** [comment]
Hi @jlajoie, for jasmine tests after updating the jasmine command the tests started failing instead of silently passing so now I've added the jasmine tests in allow_failures in Travis. You should keep the jasmine test in the GitHub actions and mark it as `continue-on-error=True`.

File: Makefile [hunk 1]
```diff
@@ -97,7 +97,7 @@ upgrade: check_pins	## update the requirements/*.txt files with the latest packa
 	$(PIP_COMPILE) -o requirements/doc.txt requirements/doc.in
 	$(PIP_COMPILE) -o requirements/test.txt requirements/test.in
 	$(PIP_COMPILE) -o requirements/dev.txt requirements/dev.in
-	$(PIP_COMPILE) -o requirements/travis.txt requirements/travis.in
+	$(PIP_COMPILE) -o requirements/github.txt requirements/github.in
```
Comment (UsamaSadiq) [inline Makefile:100]: Just a suggestion, In all other services we use `github.in` for github dependencies and have switched the name `travis.in` to `ci.in` & `ci.txt`.

**Review by taliaedX — APPROVED** [review]
I can't comment on whether the exact commands are the right syntax or not but the steps themselves and alterations from Travis to GitHub actions looks good to me.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    branches: [master]
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.5']
+        django-version: ['django22']
+        toxenv: ['django22', 'quality', 'docs', 'pii-annotations']
```
Comment (UsamaSadiq) [inline .github/workflows/ci.yml:16]: I think you should add jasmine tests too in the tox environments and mark them as allow_failures.

Comment (jlajoie) [reply .github/workflows/ci.yml:16]: Added it as a ticket after talking with George about doing this as follow up work. Would rather not have things running that have no intention of passing. Can be tracked at https://openedx.atlassian.net/browse/ENT-3880

**Pull request merged**